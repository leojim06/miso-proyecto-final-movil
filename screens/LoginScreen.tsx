import React, { useState } from 'react';
import { Block, ModalPanel, Text, Spinner } from '../components';
import LoginForm from '../components/forms/loginForm';
import { useTheme, useTranslation, useData } from '../hooks';
import useLoginEndpoint from '../services/api/useLoginEndpoint';

export default function LoginScreen() {
    const [showModal, setModal] = useState(false);
    const [error, setError] = useState();
    const { handleUser, handleLoading } = useData();
    const { colors, sizes } = useTheme();
    const { t } = useTranslation();
    const { loadLogin } = useLoginEndpoint();

    const handleSubmit = async (values: any) => {
        handleLoading(true);
        try {
            const response = await loadLogin({
                username: values.email,
                password: values.password,
            });
            handleUser(response);
        } catch (error) {
            setError(error);
            setModal(true);
        } finally {
            handleLoading(false);
        }
    };

    return (
        <Block safe>
            <Block flex={1} padding={sizes.md} color={colors.card} keyboard>
                <Text h3>{t('login.label.title')}</Text>
                <LoginForm onSubmit={(values) => handleSubmit(values)} />
            </Block>
            <ModalPanel visible={showModal} closeModal={() => setModal(false)}>
                <Text h5>{error}</Text>
            </ModalPanel>
        </Block>
    );
}
