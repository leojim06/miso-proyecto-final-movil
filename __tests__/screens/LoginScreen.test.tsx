// import AsyncStorage from '@react-native-async-storage/async-storage';
import React from 'react';
import renderer from 'react-test-renderer';

import LoginScreen from '../../screens/LoginScreen';

describe('<LoginScreen />', async () => {
    // expect(AsyncStorage.getItem).toBeCalledWith('isDark');
    it('has 2 child', () => {
        const tree: any = renderer.create(<LoginScreen />).toJSON();
        expect(tree?.children?.length).toBe(2);
    });
    it('App renders without crashing', () => {
        const rendered = renderer.create(<LoginScreen />).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('App test against snapshot', () => {
        const tree = renderer.create(<LoginScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

