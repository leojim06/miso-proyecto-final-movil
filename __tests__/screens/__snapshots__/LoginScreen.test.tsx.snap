// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> App test against snapshot 1`] = `
<View>
  <Text>
    Hola
  </Text>
  <Text>
    Mundo
  </Text>
</View>
`;
